src/openassetio-core/src/hostApi/Manager.cpp

```diff
@@ -37,7 +39,34 @@ void throwFromBatchElementError(std::size_t index, BatchElementError error) {
   }
 }
 
+/**
+ * Extract the entity reference prefix from a manager plugin's info
+ * dictionary, if available.
+ */
+std::optional<Str> entityReferencePrefixFromInfo(const log::LoggerInterfacePtr &logger,
+                                                 const InfoDictionary &info) {
+  // Check if the info dict has the prefix key.
+  if (auto iter = info.find(Str{constants::kInfoKey_EntityReferencesMatchPrefix});
+      iter != info.end()) {
+    if (const auto *prefixPtr = std::get_if<openassetio::Str>(&iter->second)) {
+      std::string msg = "Entity reference prefix '";
```

Comment:
[question] It would be good to avoid constructing this string if the logger severity is such that it won't get logged.

We could
* Merge `SeverityFilter` with `LoggerInterface` so that we can check logger severity on any  `LoggerInterface`.
* Have `HostSession` require a `SeverityFilter` as its logger rather than the more generic `LoggerInterface`.
* `dynamic_cast<SeverityFilter>` and log regardless if the cast fails, but use the severity level if the cast succeeds.

I'm not too concerned about this particular function, since it's very rarely called.  But could add the `dynamic_cast<SeverityFilter>` option if it seems worthwhile?

Reply:
That third option should really be factored into a common utility, e.g. `static` function on `SeverityFilter`.

```
bool SeverityFilter::isSeverityLogged(const LoggerInterfacePtr& logger, const Severity severity) {
  if (auto* severityFilteredLogger = dynamic_cast<SeverityFilter*>(logger.get())) {
    return severity >= severityFilteredLogger->getSeverity();
  }

  return true;
}
```

Which would then need its own tests and Python bindings etc etc, so I don't propose to do as part of this PR.

Reply:
`SeverityFilter`/`ConsoleLogger` are just conveniences. Most real-world applications bridge these methods into their own logging systems, which have their own build-in severity controls, so I think you'd have to go with the something like the first option, and also add a `Logger` wrapper that anyone logging a message uses. Plus ask host authors to update its severity when the user changes the presented severity?

Reply:
> also add a Logger wrapper that anyone logging a message uses. Plus ask host authors to update its severity when the user changes the presented severity?

I'm not clear on why these are required?

Would adding a single method to `LoggerInterface` do the trick? 
```
virtual bool isSeverityLogged(Severity severity) { return true; }
```
which should work since the `Severity` enum is a child of the `LoggerInterface`.

Reply:
Oh, sorry yes, I see what you mean, yeah if we're happy that calling back into the host to ask this needs to be as cheap as possible. I guess we'd at least want to wrap it up into macros (or something) that get used instead of the host session methods then too, as for it to work properly at saving the string overheads, no code should ever call any of the log methods directly? Just wondering if in that case, we don't want to leave them lying around in such a prominent place?

Reply:
As discussed offline, we should add a convenience macro too, but the limitation of the macro is it's a one-liner, whereas constructing a log string can span multiple lines, so in general both patterns will be required.

Reply:
[plan] Would you mind making an issue please? And we can pick this up in another PR.

Reply:
Added as https://github.com/OpenAssetIO/OpenAssetIO/issues/1014